chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/curesStatistics/CuresCriteriaStatisticsByAcbCalculator.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    public boolean hasStatisticsForDate(LocalDate statisticDate) {
+        List<CuresCriteriaStatisticsByAcb> statisticsForDate = curesStatisticsByAcbDAO.getStatisticsForDate(statisticDate);
+        return statisticsForDate != null && statisticsForDate.size() > 0;
+    }
+
+    public void deleteStatisticsForDate(LocalDate statisticDate) {
+        List<CuresCriteriaStatisticsByAcb> statisticsForDate = curesStatisticsByAcbDAO.getStatisticsForDate(statisticDate);
+        for (CuresCriteriaStatisticsByAcb statistic : statisticsForDate) {
+            try {
+                curesStatisticsByAcbDAO.delete(statistic.getId());
+            } catch (Exception ex) {
+                LOGGER.error("Could not delete statistic with ID " + statistic.getId());
+            }
+        }
+    }
+
+    public void save(List<CuresCriteriaStatisticsByAcb> statistics) {
+        statistics.stream()
+            .forEach(stat -> stat.setLastModifiedUser(User.SYSTEM_USER_ID));
+
+        curesStatisticsByAcbDAO.create(statistics);
+    }
+
+    private Long calculateUpgradeCriterionByAcb(Long acbId, List<CertifiedProductDetailsDTO> certifiedProductDetails, CertificationCriterion criterion) {
+        return certifiedProductDetails.stream()
+                .filter(listing -> listing.getCertificationBodyId().equals(acbId))
+                .filter(listing -> activityStatisticsHelper.didListingRemoveAttestationToCriterionDuringTimeInterval(listing.getId(), criterion, curesEffectiveDate, new Date()))
+                .collect(Collectors.counting());
+    }
+
+    private Long calculateNewCriterionByAcb(Long acbId, List<CertifiedProductDetailsDTO> certifiedProductDetails, CertificationCriterion criterion) {
+        return certifiedProductDetails.stream()
+                .filter(listing -> listing.getCertificationBodyId().equals(acbId))
+                .filter(listing -> !activityStatisticsHelper.didListingRemoveAttestationToCriterionDuringTimeInterval(listing.getId(), criterion, curesEffectiveDate, new Date()))
+                .collect(Collectors.counting());
+    }
+
+    private Long calculateCriteriaNeedingUpgradeCountByAcb(Long acbId, CertificationCriterion criterion) {
+        return certifiedProductDAO.getListingsAttestingToCriterion(criterion.getId(), activeStatuses).stream()
+                .filter(listing -> listing.getCertificationBodyId().equals(acbId))
+                .collect(Collectors.counting());
+    }
+
+    private List<Long> getListOfAcbs(List<CertifiedProductDetailsDTO> certifiedProductDetails) {
+        return StreamEx.of(certifiedProductDetails).map(det -> det.getCertificationBodyId()).distinct().toList();
+    }
+
+    private CertificationBody getCertificationBody(Long acbId) {
+        try {
+            return new CertificationBody(certificationBodyDAO.getById(acbId));
+        } catch (EntityRetrievalException e) {
+            return null;
+        }
+    }
+    class CuresCriteriaUpdate {
```

Comment:
Could you just put the `@Data` annotation (and maybe one or more of the constructor ones) here? Does that even work?